pkg/config/template.go

```diff
@@ -47,3 +62,68 @@ func getTemplates(templatePath string) (CiTemplates, error) {
 func isYAML(file string, info os.FileInfo) bool {
 	return !info.IsDir() && (filepath.Ext(file) == ".yaml" || filepath.Ext(file) == ".yml")
 }
+
+// TemplateCMController holds the details needed for the configmap controller
+type TemplateCMController struct {
+	cmclient  corev1.ConfigMapInterface
+	prNumber  int
+	logger    *logrus.Entry
+	templates map[string]*templateapi.Template
+}
+
+// NewTemplateCMController creates a new TemplateCMController
+func NewTemplateCMController(cmclient corev1.ConfigMapInterface, prNumber int, logger *logrus.Entry, templates map[string]*templateapi.Template) *TemplateCMController {
+	return &TemplateCMController{
+		cmclient:  cmclient,
+		prNumber:  prNumber,
+		logger:    logger,
+		templates: templates,
+	}
+}
+
+// CreateCMTemplates creates configMaps for all the changed templates.
+func (c *TemplateCMController) CreateCMTemplates() error {
+	var errors []error
+	s := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+
+	for filename, template := range c.templates {
+		buf := new(bytes.Buffer)
+
+		err := s.Encode(template, buf)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+		cmName := GetTempCMName(c.prNumber, template.Name)
+		cm := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: cmName,
+				Labels: map[string]string{
+					"ci.openshift.org/rehearse": "rehearse-temporary-cm",
+				},
+			},
+			Data: map[string]string{filename: buf.String()},
+		}
+
+		c.logger.WithField("template-name", template.Name).Info("creating rehearsal configMap for template")
+		if _, err := c.cmclient.Create(cm); err != nil && !kerrors.IsAlreadyExists(err) {
+			errors = append(errors, err)
+		}
+	}
+	return kutilerrors.NewAggregate(errors)
+}
+
+// CleanupCMTemplates deletes all the configMaps that have been created for the changed templates.
+func (c *TemplateCMController) CleanupCMTemplates() error {
+	c.logger.Info("deleting temporary template configMaps")
+	if err := c.cmclient.DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{
+		LabelSelector: fields.Set{"ci.openshift.org/rehearse": "rehearse-temporary-cm"}.AsSelector().String()}); err != nil {
```

Comment:
This would delete even CMs created by rehearsals for a different PR (see https://github.com/openshift/ci-operator-prowgen/pull/100/files#diff-33dd159268780d36aa2b7bd9253f7a29R102)

Reply:
Yes I will update that.